### PR TITLE
Use generator to enable ostree-remount.service and ostree-finalize-staged.path

### PR DESCRIPTION
**Pull request by cgwalters**
We struggled for a long time with enablement of our "internal units",
trying to follow the philosophy that units should only be enabled
by explicit preset.

See https://bugzilla.redhat.com/show_bug.cgi?id=1451458
and https://github.com/coreos/rpm-ostree/pull/1482
etc.

And I just saw chat (RH internal on a proprietary system sadly) where
someone hit `ostree-remount.service` not being enabled in CentOS8.

Thinking about this more, I realized we've shipped a systemd generator
for a long time and while its only role until now was to generate `var.mount`,
but by using it to force on our internal units, we don't require
people to deal with presets anymore.

Basically we're inverting things so that "if ostree= is on the kernel
cmdline, then enable our units" and not "enable our units, but have
them use ConditionKernelCmdline=ostree to skip".

Drop the weird gyrations we were doing around `ostree-finalize-staged.path`
too; forking `systemctl start` is just asking for bugs.

So after this, hopefully we won't ever again have to think about
distribution presets and our units.

### CODE REVIEW

**Review by jlebon — APPROVED** [review]

File: configure.ac [hunk 1]
```diff
@@ -499,6 +499,7 @@ AS_IF([test "x$with_libsystemd" = "xyes"], [
               [with_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)])
   AS_IF([test "x$with_systemdsystemunitdir" != "xno"], [
     AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])
+    AC_DEFINE_UNQUOTED([SYSTEM_DATA_UNIT_PATH], ["$with_systemdsystemunitdir"], ["unit path"])
```
Comment (dbnicholson) [inline configure.ac:502]: I think you want this outside the `AS_IF` since otherwise `SYSTEM_DATA_UNIT_PATH` might not be defined. Building in a (maybe too minimal) container for #2379:
```
../src/libostree/ostree-impl-system-generator.c: In function 'require_internal_units':
../src/libostree/ostree-impl-system-generator.c:131:18: error: 'SYSTEM_DATA_UNIT_PATH' undeclared (first use in this function)
```
Or maybe you need some `#ifdef` in the code or not to build the generator in this case.

Comment (dbnicholson) [reply configure.ac:502]: I filed #2380 to fix this.
